indexer/packages/postgres/src/stores/affiliate-info-table.ts

```diff
@@ -94,6 +97,165 @@ export async function findById(
+      SUM(affiliate_fills."price" * affiliate_fills."size") AS "referredTotalVolume"
+  FROM
+      affiliate_fills
+  GROUP BY
+      affiliate_fills."affiliateAddress"
+),
+
+-- Prepare to update affiliate_info
+-- STEP 3a: Left join affiliate_stats onto affiliate_metadata. affiliate_stats only has values for
+-- addresses with fills in the time window
+affiliate_info_update AS (
+  SELECT
+    affiliate_metadata."affiliateAddress",
+    affiliate_metadata."totalReferredUsers",    
+    affiliate_metadata."firstReferralBlockHeight",
+    COALESCE(affiliate_stats."totalReferredFees", 0) AS "totalReferredFees",
+    COALESCE(affiliate_stats."affiliateEarnings", 0) AS "affiliateEarnings",
+    COALESCE(affiliate_stats."referredNetProtocolEarnings", 0) AS "referredNetProtocolEarnings",
+    COALESCE(affiliate_stats."referredMakerTrades", 0) AS "referredMakerTrades",
+    COALESCE(affiliate_stats."referredTakerTrades", 0) AS "referredTakerTrades",
+    COALESCE(affiliate_stats."referredTotalVolume", 0) AS "referredTotalVolume"
+  FROM
+    affiliate_metadata
+  LEFT JOIN
+    affiliate_stats
+  ON affiliate_metadata."affiliateAddress" = affiliate_stats."affiliateAddress"
+)
+
+-- Step 3b: Update/upsert the affiliate info table with the new stats
+INSERT INTO affiliate_info (
+    "address", 
+    "totalReferredUsers",
+    "firstReferralBlockHeight",
+    "affiliateEarnings", 
+    "referredMakerTrades", 
+    "referredTakerTrades", 
+    "totalReferredFees", 
+    "referredNetProtocolEarnings",
+    "referredTotalVolume"
+)
+SELECT
+    "affiliateAddress",
+    "totalReferredUsers",
+    "firstReferralBlockHeight",
+    "affiliateEarnings",
+    "referredMakerTrades",
+    "referredTakerTrades",
+    "totalReferredFees",
+    "referredNetProtocolEarnings",
+    "referredTotalVolume"
+FROM 
+    affiliate_info_update
+ON CONFLICT ("address")
+DO UPDATE SET
+    "totalReferredUsers" = EXCLUDED."totalReferredUsers",
+    "firstReferralBlockHeight" = EXCLUDED."firstReferralBlockHeight",
+    "affiliateEarnings" = affiliate_info."affiliateEarnings" + EXCLUDED."affiliateEarnings",
+    "referredMakerTrades" = affiliate_info."referredMakerTrades" + EXCLUDED."referredMakerTrades",
+    "referredTakerTrades" = affiliate_info."referredTakerTrades" + EXCLUDED."referredTakerTrades",
+    "totalReferredFees" = affiliate_info."totalReferredFees" + EXCLUDED."totalReferredFees",
+    "referredNetProtocolEarnings" = affiliate_info."referredNetProtocolEarnings" + EXCLUDED."referredNetProtocolEarnings",
+    "referredTotalVolume" = affiliate_info."referredTotalVolume" + EXCLUDED."referredTotalVolume";
+
+-- Step 5: Upsert new affiliateInfoLastUpdateTime to persistent_cache table
+INSERT INTO persistent_cache (key, value)
+VALUES ('${PersistentCacheKeys.AFFILIATE_INFO_UPDATE_TIME}', '${windowEndTs}')
+ON CONFLICT (key) 
+DO UPDATE SET value = EXCLUDED.value;
+
+COMMIT;
+    `,
+  );
+}
```

Comment:
Do you have any idea of the runtime of this SQL statement? I understand that half of the tables do not exist, but you should be able to test the speed of the fills table statements and joins to ensure that the runtime is acceptable. At the very least, every field you are filtering on should have an index, because these are big tables

Reply:
This was tested during wallet total update on imperator db. For a day it is <1s.